afs-storage-api/src/main/java/com/powsybl/afs/storage/AppStorageArchive.java

```diff
@@ -211,30 +248,102 @@ public void archive(String nodeId, Path parentDir) {
         Objects.requireNonNull(parentDir);
 
         try {
-            archive(storage.getNodeInfo(nodeId), parentDir);
+            archive(storage.getNodeInfo(nodeId), parentDir, new ArchiveContext(false));
         } catch (IOException e) {
             throw new UncheckedIOException(e);
         }
     }
 
-    public void archive(NodeInfo nodeInfo, Path parentDir) throws IOException {
+    public void archive(String nodeId, Path parentDir, boolean archiveDependencies) {
+        Objects.requireNonNull(nodeId);
+        Objects.requireNonNull(parentDir);
+
+        try {
+            archive(storage.getNodeInfo(nodeId), parentDir, new ArchiveContext(archiveDependencies));
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    public void archive(NodeInfo nodeInfo, Path parentDir, ArchiveContext archiveDependencies) throws IOException {
         Objects.requireNonNull(nodeInfo);
         Objects.requireNonNull(parentDir);
+        Path directory = parentDir;
 
-        LOGGER.info("Archiving node {} ({})", nodeInfo.getId(), nodeInfo.getName());
+        if (!archiveDependencies.getIdListObject().contains(nodeInfo.getId())) {
```

Comment:
so, if a child is a dependency of a previously archived node, it won't be archived ?
for instance if in a directory we have 
- childA
- childB

and childB is a dependency of childA, then the result archive will have childB archived as a dependency and not has a child of the directory.
Maybe ArchiveContext could hold all the id+path of the dependencies gathered along the archiving process and archived at the very end (the head of the archive recursive stack) ?

Reply:
ok so, this is just internal to the archive data, when unarchived all the children are correctly unarchived, right ?